build/common.go

```diff
@@ -24,8 +27,11 @@ import (
 	"github.com/grafana/grafana-plugin-sdk-go/internal"
 )
 
-var defaultOutputBinaryPath = "dist"
-var defaultPluginJSONPath = "src"
+var (
+	defaultOutputBinaryPath     = "dist"
+	defaultPluginJSONPath       = "src"
+	defaultNestedDataSourcePath = "datasource"
```

Comment:
I have opted to not add a config option for this because I think if people want to do something out of the standard we are offering (docs to come in plugin-tools), then they can do so with mage. For example:

```go
func Foo() {
	configCallback := func(cfg build.Config) (build.Config, error) {
		cfg.PluginJSONPath = "src/myCustomPath"
		return cfg, nil
	}
	build.SetBeforeBuildCallback(configCallback)
	build.BuildAll()
}
```